media/js/base/search-params.js

```diff
@@ -42,3 +42,37 @@ _SearchParams.prototype.toString = function () {
         return [encodeURIComponent(key), encodeURIComponent(value)].join('=');
     }).join('&');
 };
+
+_SearchParams.prototype.utmParams = function() {
+    var utms = {};
+
+    $.each(this.params, function (key, val) {
+        if (key.indexOf('utm_') === 0) {
+            utms[key] = val;
+        }
+    });
+
+    return utms;
+};
+
+_SearchParams.prototype.utmParamsFxA = function(pathname) {
+    pathname = pathname || window.location.pathname || '';
+
+    var utms = this.utmParams();
+
+    // set to default value if not specified in URL
+    if (!utms.utm_campaign) {
+        utms.utm_campaign = 'page referral - not part of a campaign';
```

Comment:
According to [the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1253120#c0), the values are encoded on the product side, so I don't think we want to preemptively encode here.